src/test/java/software/amazon/encryption/s3/S3EncryptionClientRangedGetCompatibilityTest.java

```diff
@@ -37,16 +41,47 @@
 public class S3EncryptionClientRangedGetCompatibilityTest {
 
     private static SecretKey AES_KEY;
+    private static KeyPair RSA_KEY_PAIR;
 
     @BeforeAll
     public static void setUp() throws NoSuchAlgorithmException {
         KeyGenerator keyGen = KeyGenerator.getInstance("AES");
         keyGen.init(256);
         AES_KEY = keyGen.generateKey();
+
+        KeyPairGenerator keyPairGen = KeyPairGenerator.getInstance("RSA");
+        keyPairGen.initialize(2048);
+        RSA_KEY_PAIR = keyPairGen.generateKeyPair();
     }
 
-    @Test
-    public void AsyncAesGcmV3toV3RangedGet() {
+    static Object[] keyMaterialProvider() {
+        return new Object[] {
```

Comment:
nit: for extra coverage, we could add KMS here as well

Reply:
Did KMS (symmetric) key...
Are there any other ones we need to test?

Reply:
I think that's sufficient, we could possibly add v2 tests to cover each of the various RSA modes, but I don't think we should block on that.